client/landing/stepper/declarative-flow/internals/steps-repository/sensei-plan/index.tsx

```diff
@@ -0,0 +1,398 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+import { useLocale } from '@automattic/i18n-utils';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useCallback, useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { useI18n } from '@wordpress/react-i18n';
+import classnames from 'classnames';
+import wpcomProxyRequest from 'wpcom-proxy-request';
+import { Plans } from 'calypso/../packages/data-stores/src';
+import formatCurrency from 'calypso/../packages/format-currency/src';
+import { SENSEI_FLOW, SenseiStepContainer } from 'calypso/../packages/onboarding/src';
+import { useSupportedPlans } from 'calypso/../packages/plans-grid/src/hooks';
+import PlanItem from 'calypso/../packages/plans-grid/src/plans-table/plan-item';
+import { useNewSiteVisibility } from 'calypso/landing/gutenboarding/hooks/use-selected-plan';
+import { PLANS_STORE } from 'calypso/landing/gutenboarding/stores/plans';
+import {
+	USER_STORE,
+	ONBOARD_STORE,
+	SITE_STORE,
+	PRODUCTS_LIST_STORE,
+} from 'calypso/landing/stepper/stores';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { domainRegistration } from 'calypso/lib/cart-values/cart-items';
+import { cartManagerClient } from 'calypso/my-sites/checkout/cart-manager-client';
+import { SenseiStepError } from '../sensei-setup/sensei-step-error';
+import { SenseiStepProgress, Progress } from '../sensei-setup/sensei-step-progress';
+import { Tagline, Title, PlansIntervalToggle } from './components';
+import type { Step } from '../../types';
+import type { StyleVariation } from 'calypso/../packages/design-picker/src/types';
+import type { Props as PlanItemProps } from 'calypso/../packages/plans-grid/src/plans-table/plan-item';
+import 'calypso/../packages/plans-grid/src/plans-table/style.scss';
+import './styles.scss';
+
+enum Status {
+	Initial,
+	Bundling,
+	Error,
+}
+
+const siteProgressTitle: string = __( 'Laying out the foundations' );
+const cartProgressTitle: string = __( 'Preparing Your Bundle' );
+const styleProgressTitle: string = __( 'Applying your site styles' );
+
+const getStyleVariations = ( siteId: number, stylesheet: string ): Promise< StyleVariation[] > =>
+	wpcomProxyRequest( {
```

Comment:
I'm not too familiar with the usage of `wpcomProxyRequest`, but I did find a P2 recommending we should not use wpcomProxyRequest directly (if I'm understanding correctly): p4TIVU-9GP-p2#comment-10997.

Im wondering if we should use wpcom.req.get / wpcom.req.post or React Query instead?

@alshakero - Do you have knowledge regarding these api requests?

Reply:
Yes I'd totally move this to either a store on a React query.

Reply:
I switched to using wpcom.req for now. We'll considering creating a store in a future PR 095f8dcec971f31d505997d3a488f88e1643aad7